core/src/main/kotlin/dev/hotwire/core/logging/CoreLog.kt

```diff
@@ -1,45 +1,32 @@
 package dev.hotwire.core.logging
 
-import android.util.Log
 import dev.hotwire.core.config.Hotwire
+import okhttp3.logging.HttpLoggingInterceptor
 
-internal object CoreLog {
-    private const val DEFAULT_TAG = "Hotwire-Core"
-
-    private val debugEnabled get() = Hotwire.config.debugLoggingEnabled
-
-    internal fun d(msg: String) = log(Log.DEBUG, msg)
-
-    internal fun w(msg: String) = log(Log.WARN, msg)
-
-    internal fun e(msg: String) = log(Log.ERROR, msg)
+private const val DEFAULT_TAG = "Hotwire-Core"
+private const val PAD_END_LENGTH = 35
 
-    private fun log(logLevel: Int, msg: String) {
-        when (logLevel) {
-            Log.DEBUG -> if (debugEnabled) Log.d(DEFAULT_TAG, msg)
-            Log.WARN -> Log.w(DEFAULT_TAG, msg)
-            Log.ERROR -> Log.e(DEFAULT_TAG, msg)
-        }
+internal class HotwireHttpLogger : HttpLoggingInterceptor.Logger {
+    override fun log(message: String) {
+        Hotwire.config.logger.d(DEFAULT_TAG, message)
     }
 }
 
-private const val PAD_END_LENGTH = 35
-
 internal fun logEvent(event: String, details: String = "") {
-    CoreLog.d("$event ".padEnd(PAD_END_LENGTH, '.') + " [$details]")
+    Hotwire.config.logger.d(DEFAULT_TAG, "$event ".padEnd(PAD_END_LENGTH, '.') + " [$details]")
```

Comment:
Can we refactor these into something more readable, like `"${event.padded()} [$details]"`? My brain no longer accepts string concatenation. 😅